e2e/cypress/integration/messaging/message_reaction_gm_spec.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod @smoke
+// Group: @messaging
+
+describe('Emoji reactions to posts/messages in GM channels', () => {
+    let userOne;
+    let userTwo;
+    let testTeam;
+    let testGroupChannel;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+            userOne = user;
+
+            cy.apiCreateUser().then((data) => {
+                userTwo = data.user;
+
+                cy.apiAddUserToTeam(testTeam.id, userTwo.id);
+
+                cy.apiCreateGroupChannel([userOne.id, userTwo.id]).then((response) => {
+                    testGroupChannel = response.body;
+                });
+
+                // # Login as userOne and town-square
+                cy.apiLogin(userOne);
+                cy.visit(`/${testTeam.name}/channels/town-square`);
+            });
+        });
+    });
+
+    it('MM-T471 add a reaction to a message in a GM', () => {
+        // # Switch to the GM
+        cy.visit(`/${testTeam.name}/messages/${testGroupChannel.name}`);
+
+        // # Post a message
+        cy.postMessage('This is a post');
+
+        // # Mouseover the last post
+        cy.getLastPost().trigger('mouseover');
+
+        cy.getLastPostId().then((postId) => {
+            // # Click the add reaction icon
+            cy.clickPostReactionIcon(postId);
+
+            // # Choose "slightly_frowning_face" emoji
+            // delaying 500ms in case of lag
+            cy.get('.emoji-picker__items #emoji-1f641').wait(500).click();
+
+            // * The number shown on the reaction is incremented by 1
+            cy.get(`#postReaction-${postId}-slightly_frowning_face .Reaction__number--display`).
+                should('have.text', '1').
+                should('be.visible');
+        });
```

Comment:
The cypress test is missing the following assertion specified in the TM4J test:

`Narrow the Desktop app window and verify plus sign appears next to existing reaction (hover not needed in narrow mobile view)`

Suggested changes:

```
            // # resize window to mobile view
            cy.viewport('iphone-6');

            // # Focus on last postReaction - Workaround for hover not needed in Narrow Desktop view
            cy.get('#post_textbox').type(' testing').focus().tab().tab({shift: true}).tab({shift: true});

            // * Verify if Add Reaction button is displayed
            cy.findByLabelText('Add a reaction').should('be.visible');
```

Reply:
Actually, this seems to be failing currently based on those steps since it says the button should appear without hovering in mobile view. Should I submit a fix for that at the same time?

Reply:
I fixed the visibility of that button. Some CSS got renamed when the animations were added, and that broke the visibility in mobile view.